google/cloud/spanner/internal/connection_impl.cc

```diff
@@ -109,9 +111,13 @@ ConnectionImpl::ConnectionImpl(
       backoff_policy_prototype_(std::move(backoff_policy)),
       background_threads_(options.background_threads_factory()()),
       session_pool_(MakeSessionPool(
-          db_, std::move(stubs), std::move(session_pool_options),
-          background_threads_->cq(), retry_policy_prototype_->clone(),
-          backoff_policy_prototype_->clone())),
+          db_, std::move(stubs), background_threads_->cq(),
+          DefaultOptions(
```

Comment:
Note: Once `ConnectionImpl` itself accepts `internal::Options` (likely the next PR), the next 5 lines will go away.